pkg/capacity/capacity_test.go

```diff
@@ -1232,6 +1245,94 @@ func fakeController(ctx context.Context, client *fakeclientset.Clientset, storag
 	return c, registry
 }
 
+// rateLimitingQueue is a stripped down implementation
+// which only supports adding and removing items.
+type rateLimitingQueue struct {
+	mutex        sync.Mutex
+	items        []interface{}
+	shuttingDown bool
+}
+
+var _ workqueue.RateLimitingInterface = &rateLimitingQueue{}
+
+func (r *rateLimitingQueue) Add(item interface{}) {
+	r.mutex.Lock()
+	defer r.mutex.Unlock()
+
+	for _, existing := range r.items {
+		if existing == item {
+			return
+		}
+	}
+	r.items = append(r.items, item)
+}
+func (r *rateLimitingQueue) Len() int {
+	r.mutex.Lock()
+	defer r.mutex.Unlock()
+
+	return len(r.items)
+}
+func (r *rateLimitingQueue) Get() (item interface{}, shutdown bool) {
+	done := func() bool {
+		r.mutex.Lock()
+		defer r.mutex.Unlock()
+
+		if len(r.items) > 0 {
+			item = r.items[0]
+			r.items = r.items[1:]
+			return true
+		}
+
+		if r.shuttingDown {
+			shutdown = true
+			return true
+		}
+		return false
+	}
+
+	for !done() {
+		time.Sleep(time.Millisecond)
```

Comment:
If we are only testing for queue content, does it need to sleep here? And could we get away a no-op shutdown?

Reply:
`Get` is never called at all during `TestRefresh`. Should I rip out the implementation entirely and replace with a panic in case that it does get called?

Reply:
Ah ok test cases use `allItems()` instead. Panicking sgtm.

Reply:
What I hadn't considered is that the same queue implementation is also used for `TestCapacity`, and there `Get` is called. Let's keep it in place.

Reply:
Does the `Get` call there depend on the delay & shutdown?